### PR TITLE
fix: colorPropToSVG comply with svg specification

### PR DESCRIPTION
**Pull request by tyutjohn**
## Motivation
@closes https://github.com/fabricjs/fabric.js/issues/9407

## Description

Change the output content of the colorPropToSVG function

## Changes
Change ':' to '='

How to write tests correctly, I need a little help, old test component seem doesn't work. This change is very small. Can anyone help me add test? I'm a little busy recently, Sorry

### CODE REVIEW

**Comment by ShaMan123** [comment]
I am guessing this is a regression, how didn't our tests catch this?

**Review by ShaMan123 — CHANGES_REQUESTED** [review]
This seems wrong to me
Looking at usages in fabric the only usage that looks wrong is 
image

**Comment by ShaMan123** [comment]
and indeed that is what you reported in your issue

**Comment by ShaMan123** [comment]
I suggest refactoring the util `colorPropToSVG` to accept a third arg `separator: ':' | '='`

**Comment by tyutjohn** [comment]
> refactoring

I don't know if it's necessary to add a third arg bacause I only found the function is used only here

**Comment by tyutjohn** [comment]
> I suggest refactoring the util `colorPropToSVG` to accept a third arg `separator: ':' | '='`

On the old code I only found a file using it, do you want to add the default value ':' to the third parameter?

**Comment by ShaMan123** [comment]
Find all usages and add what is needed IMO, very little usage.

**Comment by tyutjohn** [comment]
> I am guessing this is a regression, how didn't our tests catch this?

How to do regression testing? The old test component should not test whether the exported svg is correct.

**Comment by asturur** [comment]
I m actually trying to figure out which part was broken and then i can help you writing a test.

File: src/util/misc/svgParsing.ts [hunk 1]
```diff
@@ -139,20 +141,20 @@ export const matrixToSVG = (transform: TMat2D) =>
...
     const color = new Color(value),
       opacity = color.getAlpha();
 
-    let str = `${prop}: ${color.toRgb()}; `;
+    let str = separator === ':' ? `${prop}: ${color.toRgb()}; ` : `${prop}="${color.toRgb()}" `;
 
     if (opacity !== 1) {
       //change the color in rgb + opacity
-      str += `${prop}-opacity: ${opacity.toString()}; `;
+      str += separator === ':' ? `${prop}-opacity: ${opacity.toString()}; ` : `${prop}-opacity="${opacity.toString()}" `;
```
Comment (asturur) [inline src/util/misc/svgParsing.ts:157]: the code is fine as it is, and i not holding an approval on this.
But you you want to step up the game of cleaning up, you can remove at least 2 ternaries with something like this:

```ts
let colorValue;
let opacityValue;
if (!value) { 
  colorValue = 'none';
} else if(...) {
  colorValue = `#SVGID_${value.id}`;
} else {
 ....
 do color value and opacity value;
}
if (separator === ':') {
  return `${prop}: ${colorValue} ${opacityValue ? `${prop}-opacity: ${opacity.toString()}; ` : ''}`
} else {
 return `${prop}="${color.toRgb()}" ${opacityValue ? `${prop}-opacity="${opacity.toString()}" ` : ''}`
}

```

this maybe is easier to read? just a suggestion follow your gut feeling is not a big deal

**Comment by asturur** [comment]
for the test, create a file

src/shapes/Text/TextSVGExportMixin.spec.ts

and inside test an svg export with background color:

```ts
describe('TextSvgExport', () => {
  it('exports text background color correctly', () => {
    const myText = new Text('This is a test', { backgroundColor: 'rgba(100, 0, 100, 0.5)' });
    const svgString = myText.toSVG();
    expect(svgString).toInclude('fill=....') // find the correct string
    expect(svgString).not.toInclude('fill: ....') // find the other string
  })
});

```

in case .toInclude is not an expect matcher, just use expect(svgString.includes(...string)).toBe(true) and toBe(false)

Repeat the same test with different color to test 'none' and the other cases with a gradient i suppose.

**Comment by asturur** [comment]
i also argue if seaparator is a good variable name.
Is kind of indirect, because the separator is the ':' for styles and '=' for attributes, but what change is not just the separator but the quotes too.
A non expert dev could say, ok the function support a separator i can add a different one, while what the function does is just  forInlineStyle true/false or forAttributes false/true. Probably a boolean is nicer than the separator string.

Not an holding issue for me, just thought it could be pointed out

**Comment by tyutjohn** [comment]
> i also argue if seaparator is a good variable name. Is kind of indirect, because the separator is the ':' for styles and '=' for attributes, but what change is not just the separator but the quotes too. A non expert dev could say, ok the function support a separator i can add a different one, while what the function does is just forInlineStyle true/false or forAttributes false/true. Probably a boolean is nicer than the separator string.
> 
> Not an holding issue for me, just thought it could be pointed out

I completely agree with you, I will improve it and add a new test file to complete the testing of the function, thanks for you help

**Comment by tyutjohn** [comment]
> > i also argue if seaparator is a good variable name. Is kind of indirect, because the separator is the ':' for styles and '=' for attributes, but what change is not just the separator but the quotes too. A non expert dev could say, ok the function support a separator i can add a different one, while what the function does is just forInlineStyle true/false or forAttributes false/true. Probably a boolean is nicer than the separator string.
> > Not an holding issue for me, just thought it could be pointed out
> 
> I completely agree with you, I will improve it and add a new test file to complete the testing of the function, thanks for you help

Work done, everything looks fine

**Comment by asturur** [comment]
We also need a new line in the changelog.md file in the root folder, sorry i forgot to ask you that

**Comment by tyutjohn** [comment]
> We also need a new line in the changelog.md file in the root folder, sorry i forgot to ask you that

Already added, please try again

**Review by asturur — APPROVED** [review]

**Comment by asturur** [comment]
Since @ShaMan123 requested some changes, let's give him time to review, but i think this is great.
Thank your for your contribution

**Review by ShaMan123 — APPROVED** [review]
Good job guys

**Pull request merged**